### PR TITLE
added new logic in sales_order and added sales_reciept module

### PR DESCRIPTION
**Pull request by pavanpulicherla18345**

### CODE REVIEW

File: config/utils_variables.py [hunk 1]
```diff
@@ -146,4 +146,19 @@
...
+default_workflow_stages = {
+    1: 'sale_order',
+    2: 'dispatch',
+    3: 'invoice'
+}
+
+# Accessing values
+first_stage = default_workflow_stages[1]  # 'sale_order'
+second_stage = default_workflow_stages[2]  # 'dispatch'
+third_stage = default_workflow_stages[3]  # 'invoice'
```
Comment (QodeFarm) [inline config/utils_variables.py:164]: These are not needed, you can access them directly oin the code. right?

File: database/cnl.sql [hunk 2]
```diff
@@ -1884,5 +1884,35 @@ CREATE TABLE IF NOT EXISTS labor (
...
+  CONSTRAINT `workflow_stages_ibfk_1` FOREIGN KEY (`workflow_id`) REFERENCES `workflows` (`workflow_id`)
+);
+ 
+CREATE TABLE `workflows` (
+  `workflow_id` char(36) NOT NULL,
+  `name` varchar(255) NOT NULL,
+  `created_at` timestamp NULL DEFAULT CURRENT_TIMESTAMP,
+  `updated_at` timestamp NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  PRIMARY KEY (`workflow_id`)
+);
```
Comment (QodeFarm) [inline database/cnl.sql:1905]: Why are  ` ` added here?

File: database/cnl.sql [hunk 3]
```diff
@@ -1884,5 +1884,35 @@ CREATE TABLE IF NOT EXISTS labor (
...
+
+/* Sale Receipts Table */
+-- Stores receipts associated with sale invoices.
+CREATE TABLE IF NOT EXISTS sale_receipts (
+    sale_receipt_id CHAR(36) PRIMARY KEY,
+    sale_invoice_id CHAR(36) NOT NULL,
+    receipt_name VARCHAR(255) NOT NULL,
```
Comment (QodeFarm) [inline database/cnl.sql:1912]: Can you aldo add column receipt_path JSON, here

File: config/utils_variables.py (hunk 1 above)
Comment (pavanpulicherla18345) [reply config/utils_variables.py:164]: yeah yes Thulasi.

File: database/cnl.sql (hunk 2 above)
Comment (pavanpulicherla18345) [reply database/cnl.sql:1905]: Actually, Thulasi, I forgot to add those two tables to the database file, so I took their structures from production and added them here.

**Pull request merged**